tests/test_bootpolicy.robot

```diff
@@ -0,0 +1,187 @@
+    Write Attribute    /org/openbmc/settings/host0   boot_policy   data=${valueDict}
+    Read the Attribute  /org/openbmc/settings/host0    boot_policy
+    Response Should Be Equal   ONETIME
+
+    Trigger Warm Reset
+ 
+    Read the Attribute  /org/openbmc/settings/host0    boot_policy
+    Response Should Be Equal   ONETIME
+
+    Read the Attribute  /org/openbmc/settings/host0    boot_flags
+    Response Should Be Equal   Network
+ 
+Permanent boot order after warm reset
+    [Documentation]   This testcase is to verify that boot policy and order does not change  
+    ...               after warm reset on a system with permanent boot policy.
+
+    Power On Host
+
+    ${bootpolicy} =   Set Variable   PERMANENT
+    ${valueDict} =   create dictionary   data=${bootpolicy}
+    Write Attribute    /org/openbmc/settings/host0   boot_policy   data=${valueDict}
+    Read the Attribute  /org/openbmc/settings/host0    boot_policy
+    Response Should Be Equal   PERMANENT
+
+    ${bootDevice} =   Set Variable   CDROM
+    ${valueDict} =   create dictionary   data=${bootDevice}
+    Write Attribute    /org/openbmc/settings/host0   boot_flags   data=${valueDict}
+    Read the Attribute  /org/openbmc/settings/host0    boot_flags
+    Response Should Be Equal   CDROM
+
+    Trigger Warm Reset
+
+    Read the Attribute  /org/openbmc/settings/host0    boot_policy
+    Response Should Be Equal   PERMANENT
+
+    Read the Attribute  /org/openbmc/settings/host0    boot_flags
+    Response Should Be Equal   CDROM
+ 
+Set boot policy to invalid value
+    [Documentation]   This testcase is to verify that proper error message is prompted  
+    ...               when invalid value to provided to boot policy.
+   
+    ${bootpolicy} =   Set Variable   abc
+    ${valueDict} =   create dictionary   data=${bootpolicy}
+    Write Attribute    /org/openbmc/settings/host0   boot_policy   data=${valueDict}
+    Read the Attribute  /org/openbmc/settings/host0    boot_policy
+    Response Should Be Equal   abc
+    
+*** Keywords ***
+
+Response Should Be Equal
+    [arguments]    ${args}
+    Should Be Equal    ${OUTPUT}    ${args}
+
+Read the Attribute     
+    [arguments]    ${uri}    ${parm}
+    ${output} =     Read Attribute      ${uri}    ${parm}
+    set test variable    ${OUTPUT}     ${output}
+
+Power On Host
+    log to console    "Powering on the host"
+    @{arglist}=   Create List
+    ${args}=     Create Dictionary    data=@{arglist}
+    ${resp}=   Call Method    /org/openbmc/control/chassis0/    powerOn    data=${args}
+    should be equal as strings      ${resp.status_code}     ${HTTP_OK}
+    ${json} =   to json         ${resp.content}
+    should be equal as strings      ${json['status']}      ok
+    sleep   30sec
+    ${currentSystemState}=      Read Attribute    /org/openbmc/settings/host0    system_state
+    Should be equal   ${currentSystemState}     HOST_POWERED_ON
+
+Power Off Host
+    log to console    "Powering off the host"
```

Comment:
I think it better to keep it here. This message will get us know that power off is in progress.